lib/trackler/problem.rb

```diff
@@ -7,81 +7,108 @@ class Problem
     def initialize(slug, root)
       @slug = slug
       @root = root
+      @xcommon_repository_url = 'https://github.com/exercism/x-common/blob/master'
     end
 
     def exists?
-      !!md && !!yaml
+      description_exists? && yaml_exists?
     end
 
     def name
-      @name ||= slug.split('-').map(&:capitalize).join(' ')
+      @name ||= slug_to_name
```

Comment:
Yeah, no reason to memoize. That's just habit on my part.

> What is the cost of having more, smaller functions?

I find that a small method that isn't particularly expressive makes the code big without making it more clear. More lines of code is a trade-off that I'm happy to make when it clearly pays off. I think I do tend to stop earlier than you. I get kind of annoyed when there are 10 or 20 extra methods that don't tell a better story.